test/tools/syntaxTestTool.cpp

```diff
@@ -0,0 +1,354 @@
+					boost::get_error_info<errinfo_sourceLocation>(*error)->start
+				);
+				if (line >= 0)
+					cout << "(" << line << "): ";
+			}
+			cout << error->typeName() << ": " << errorMessage(*error) << color::reset() << endl;
+		}
+}
+
+bool SyntaxTestTool::process()
+{
+	bool match = true;
+	bool parserError = false;
+	ErrorList errorList;
+
+	m_test = SyntaxTestParser().parse(m_path.string());
+
+	try
+	{
+		errorList = parseAnalyseAndReturnError(m_test.source, true, true, true).second;
+
+		if (errorList.size() != m_test.expectations.size())
+			match = false;
+		else
+		{
+			for (size_t i = 0; i < errorList.size(); i++)
+			{
+				if (!(errorList[i]->typeName() == m_test.expectations[i].type) ||
+					!(errorMessage(*errorList[i]) == m_test.expectations[i].message))
+				{
+					match = false;
+					break;
+				}
+			}
+		}
+	}
+	catch (...)
+	{
+		match = false;
+		parserError = true;
+		errorList = m_compiler.errors();
+	}
+
+	cout << color::bold() << m_name << ": " << color::reset();
+
+	if (match)
+		cout << color::green() << "OK" << color::reset() << endl;
+	else
+	{
+		cout << color::red() << "FAILED" << color::reset() << endl;
+
+		cout << "  Contract:" << endl;
+		printContract();
+
+		if (parserError)
+		{
+			cout << "  " << color::inverse() << color::red() << "Parsing failed:" << color::reset() << endl;
+			printErrorList(errorList, true);
+		}
+		else
+		{
+			cout << "  Expected result:" << endl;
+			printExpected();
+			cout << "  Obtained result:" << endl;
+			printErrorList(errorList, false);
+		}
+		cout << endl;
+
+		if (parserError)
+			cout << "(e)dit/(s)kip/(q)uit? ";
+		else
+			cout << "(e)dit/(r)eplace/(s)kip/(q)uit? ";
+		cout.flush();
+
+		bool done = false;
+		while (!done)
+		{
+			switch(cin.get())
+			{
+				case 's':
+					done = true;
+					break;
+				case 'r':
+					if (!parserError)
+					{
+						ofstream file(m_path.string(), ios::trunc);
+
+						file << m_test.source;
+						file << "// ----" << endl;
+						for (auto &error: errorList)
+							file << "// " << error->typeName() << ": " << errorMessage(*error) << endl;
+
+						done = true;
+					}
+					break;
+				case 'e':
+					cout << endl << endl;
+					(void)system((editor + " " + m_path.string()).c_str());
+					return process();
+				case 'q':
+					cout << endl;
+					return false;
+				default:
+					break;
+			}
+		}
+		cout << endl << endl;
+	}
+	return true;
+
+}
+
+bool SyntaxTestTool::processPath(fs::path const& _basepath, fs::path const& _path)
+{
+	fs::path fullpath = _basepath / _path;
+	if (fs::is_directory(fullpath))
+	{
+		for (auto const& entry: fs::directory_iterator(fullpath))
+			if (!processPath(_basepath, _path / entry.path().filename()))
+				return false;
+	}
+	else
+	{
+		SyntaxTestTool testTool(_path.string(), fullpath);
+		if (!testTool.process())
+			return false;
+	}
+	return true;
+}
+
+int main(int argc, char *argv[])
+{
+	DisableConsoleBuffering disableConsoleBuffering;
+	(void)disableConsoleBuffering;
```

Comment:
Or any coding style preference for unused variables in a case like this (scoped setup and cleanup in constructor and descructor)?

Reply:
Is it marked as unused by the compiler if the constructor has side-effects?